src/form.js

```diff
@@ -133,34 +133,51 @@ define([
         this.enableInstanceRefCounting = opts.enableInstanceRefCounting;
         this.errors = [];
         this.question_counter = 1;
-        this.xpath = xpath.createParser(xpath.makeXPathModels(this.hashtagDictionary));
+        this.xpath = bananas.Parser(this.hashtagDictionary);
 
         //make the object event aware
         util.eventuality(this);
     }
 
     Form.prototype = {
-        addHashtag: function(hashtag, xpath) {
-            this.hashtagDictionary[hashtag] = xpath;
+        validHashtags: function() {
+            return _.keys(this.hashtagDictionary);
```

Comment:
I think this should just return `this.hashtagDictionary` so the code that references it can do a hash lookup rather than a linear search through an array. Better yet, change it to
```js
isValidHashtag: function(tag) {
    return this.hashtagDictionary.hasOwnProperty(tag);
}
```